### PR TITLE
Task/dr 369 augment git role

### PR DESCRIPTION
**Pull request by timrickwood**
## Description
Previous workflows to run terraform changes failed for missing permissions 
Also took opportunity to add required cloudshell permission to developer role 

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by jope1 — APPROVED** [review]

File: infrastructure/stacks/uec-developer-role/uec-rw-data-services.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
                 "redshift:DescribeEvents",
                 "redshift-serverless:ListWorkgroups",
                 "redshift:CreateSavedQuery",
+                "backup:*",
```
Comment (jope1) [inline infrastructure/stacks/uec-developer-role/uec-rw-data-services.json:31]: Might be my misunderstanding, but I was expecting to see these permissions:
"dynamodb:CreateBackup",
"dynamodb:RestoreTableFromBackup",

However it looks like those should already be covered by the 'VisualEditor4' policy below (line 117)

File: infrastructure/stacks/uec-developer-role/uec-rw-data-services.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
                 "redshift:DescribeEvents",
```
Comment (jope1) [inline infrastructure/stacks/uec-developer-role/uec-rw-data-services.json:28]: A bit off topic, but I didn't realise we were using redshift? Isn't that a data warehousing component?

Doesn't the 'VisualEditor2' policy override this one here anyway?

Comment (timrickwood) [reply infrastructure/stacks/uec-developer-role/uec-rw-data-services.json:28]: ooh good spot .. the json was lifted from the console so not quite sure how the redundant redshift perms got included, I will edit it down to just use redshift.* ... I think as this is intended for use by devs in dev account only I think we included resources we might want to play with in future.

File: infrastructure/stacks/uec-developer-role/uec-rw-data-services.json (hunk 1 above)
Comment (timrickwood) [reply infrastructure/stacks/uec-developer-role/uec-rw-data-services.json]: Possible duplicate reply. Those permissions are covered by dynamodb.* but by themselves they were not sufficient - hence the blanket inclusion of backup

**Review by Pira-nhs — APPROVED** [review]

**Review by tamaragoldschmidt — APPROVED** [review]

**Pull request merged**